File: FrontEnd/src/services/UserService.js
import axios from 'axios';

const BASE_PATH_URL="http://localhost:8080/User";

class UserService{
    
    userLogin(email,password){
        return axios.post(BASE_PATH_URL+`/login/${email}/${password}`)
    }

    getAllUsers(){
    return axios.get(BASE_PATH_URL+"/getAllUsers")
    }

   getAllPatients(){
    return axios.get(BASE_PATH_URL+"/getAllPatients");
   }

   addPatient(name,age,gender,contact,medicalhistory){
    return axios.post(BASE_PATH_URL+"/addPatient",{name,age,gender,contact,medicalhistory})
   }

   updatePatient(patientId,name,age,gender,contact,medicalhistory){
    return axios.put(BASE_PATH_URL+`/Patient/update/${patientId}`,{name,age,gender,contact,medicalhistory})
   }
   
   deletePatient(id){
    return axios.delete(BASE_PATH_URL+`/Patient/delete/${id}`)
   }

   userLogout(userId){
    return axios.get(BASE_PATH_URL+`/logout/${userId}`)
   }

   getAllReports(){
    return axios.get(BASE_PATH_URL+"/getAllReports");
   }

   addReport(userId,patientId,testResults,diagnosisInfo,date){
    return axios.post(BASE_PATH_URL+"/addReport",{userId,patientId,testResults,diagnosisInfo,date})
   }

   updateReport(reportId,userId,patientId,testResults,diagnosisInfo,date){
    return axios.put(BASE_PATH_URL+`/Report/update/${reportId}`,{userId,patientId,testResults,diagnosisInfo,date})
   }

   deleteReport(id){
    return axios.delete(BASE_PATH_URL+`/Report/delete/${id}`)
   }

   getReportById(newid){
    return axios.get(BASE_PATH_URL+`/Report/getbyId/${newid}`)
   }

   getAllAppointments(){
    return axios.get(BASE_PATH_URL+"/getAllAppointments")
   }

   addAppointment(uid,pid,date){
    return axios.post(BASE_PATH_URL+"/request",{uid,pid,date})
   }

   getAppointmentById(newid){
    return axios.get(BASE_PATH_URL+`/Appointment/getbyId/${newid}`)
   }

   getMyAppRequests(newid){
    return axios.get(BASE_PATH_URL+`/getMyRequests/${newid}`)
   }

   acceptRequest(newid){
    return axios.get(BASE_PATH_URL+`/acceptRequest/${newid}`)
   }

}

export default UserService

File: BackEnd/src/main/java/com/manipal/demo/service/UserService.java
package com.manipal.demo.service;

import java.util.List;


import java.util.Optional;
import org.springframework.stereotype.Service;

import com.manipal.demo.dto.AptDto;
import com.manipal.demo.dto.ReportDto;
import com.manipal.demo.dto.Request;
import com.manipal.demo.entity.Appointment;
import com.manipal.demo.entity.Patient;
import com.manipal.demo.entity.Report;
import com.manipal.demo.entity.User;
@Service
public interface UserService {
	
   public User userLogin(String email, String password);
   public String userLogout(Long id);
   public User userRegister(User user);
   public Optional<Patient> findPatientById(long id);
   //public Optional<Patient> findPatientByName(String name);
   public List<Patient> getAllPatients();
   public List<User> getAllUsers();
 
   public void addPatient(Patient patient);
   public void deletePatientById(long id);
   public void updatePatient(Long id,Patient patient);
   
   public List<Report> getAllReports();
   public String addReport(ReportDto reportDto);
   public List<Report> findReportByUserName(String name);
   public Optional<Report> findByPatientId(long id);
   
   public void deleteReportById(long id);
   public void updateReport(Long r_id,ReportDto reportDto);
   
   public String addAppointment(AptDto aptdto);
   public void updateAppointment(Long apt_id,AptDto aptdto);
   public List<Appointment> getAllAppointments();
   public List<Appointment> findAppointmentById(long id);
   
  
   public String acceptRequest(Request req);
   
}
